lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3848,65 +3852,72 @@ ParseResult FIRCircuitParser::parseCircuit(
   std::string circuitTarget = "~" + name.getValue().str();
   size_t nlaNumber = 0;
 
-  ArrayAttr annotations;
-  if (getConstants().options.rawAnnotations) {
-    SmallVector<Attribute> rawAnno;
-    // Deal with any inline annotations, if they exist.  These are processed
-    // first to place any annotations from an annotation file *after* the inline
-    // annotations.  While arbitrary, this makes the annotation file have
-    // "append" semantics.
-    if (!inlineAnnotations.empty())
-      if (importAnnotationsRaw(inlineAnnotationsLoc, circuitTarget,
-                               inlineAnnotations, rawAnno))
-        return failure();
+  // Create a scope to observe elapsed time.
+  {
+    auto parseAnnotationTimer = ts.nest("Parse annotations");
```

Comment:
Doing this via RAII/relying on the destructor to stop the timer might be overly complex. Is it simpler to just create the nested timing scope and then stop it at the bottom?  I'd just see what it looks like and make a call. I'm fine with any solution.